### PR TITLE
celldb in-memory mode, stats for actors, perf counters, minor fix in rldp2

### PR DESCRIPTION
**Pull request by birydrad**

### CODE REVIEW

File: crypto/vm/db/InMemoryBagOfCellsDb.cpp [hunk 1]
```diff
@@ -0,0 +1,982 @@
...
+            mmap(NULL, batch_size, PROT_READ | PROT_WRITE, MAP_PRIVATE | MAP_ANONYMOUS, -1, 0));
+        CHECK(ptr != nullptr);
+        return std::unique_ptr<char, Deleter>(ptr);
+      }
+      void operator()(char *ptr) const {
+        munmap(ptr, batch_size);
+      }
+#else
+      static std::unique_ptr<char, Deleter> alloc() {
+        auto ptr = reinterpret_cast<char *>(malloc(batch_size)));
```
Comment (Lordron) [inline crypto/vm/db/InMemoryBagOfCellsDb.cpp]: excessive ) symbol

File: tdactor/td/actor/core/ActorTypeStat.cpp [hunk 2]
```diff
@@ -0,0 +1,109 @@
+#include "td/actor/core/Actor.h"
+#include "td/actor/core/ActorTypeStat.h"
+#include "td/actor/core/Scheduler.h"
+#include "td/utils/port/thread_local.h"
+#include <cxxabi.h>
```
Comment (Lordron) [inline tdactor/td/actor/core/ActorTypeStat.cpp]: not available under windows

File: tdutils/td/utils/port/Clocks.h [hunk 3]
```diff
@@ -26,6 +27,62 @@ struct Clocks {
   static double system();
 
   static int tz_offset();
+
+#if defined(__i386__)
```
Comment (Lordron) [inline tdutils/td/utils/port/Clocks.h]: all of these defines are not available under windows and MSVC. On windows you can use __rdtsc and __forceinline (__inline__ also seems missing). Look and libmzp rdtsc implementation

**Review by EmelyanenkoK — APPROVED** [review]

File: tdutils/td/utils/port/Clocks.h (hunk 3 above)
Comment (moodarkred) [reply tdutils/td/utils/port/Clocks.h]: Hi

**Review by MrG9090 — APPROVED** [review]

**Pull request merged**